pyfr/plugins/turbulence.py

```diff
@@ -0,0 +1,226 @@
+import math
+import numpy as np
+
+from collections import defaultdict
+from pyfr.plugins.base import BaseSolverPlugin
+from pyfr.regions import BoxRegion
+from rtree.index import Index, Property
+
+
+class pcg32rxs_m_xs:
+    def __init__(self, seed):
+        self.state = np.uint32(seed)
+    def random(self):
+        oldstate = np.uint32(self.state)
+        self.state = (oldstate * np.uint32(747796405)) + np.uint32(2891336453)
+        rshift = np.uint32(oldstate >> np.uint32(28))
+        oldstate ^= oldstate >> (np.uint32(4) + rshift)
+        oldstate *= np.uint32(277803737)
+        oldstate ^= oldstate >> np.uint32(22)
+        return (oldstate >> 8) * 5.9604644775390625e-8
+    def getstate(self):
+        return self.state
+
+
+class TurbulencePlugin(BaseSolverPlugin):
+    name = 'turbulence'
+    systems = ['navier-stokes']
+    formulations = ['dual', 'std']
+    dimensions = [3]
+
+    def __init__(self, intg, cfgsect):
+        super().__init__(intg, cfgsect)
+
+        fdptype = intg.backend.fpdtype
+
+        self.tstart = intg.tstart
+        self.tbegin = intg.tcurr
+        self.tnext = intg.tcurr
+        self.tend = intg.tend
+
+        self.seed = self.cfg.getint(cfgsect, 'seed')
+
+        self.vortbuffdtype = np.dtype([('yinit', fdptype), ('zinit', fdptype),
+                                   ('tinit', fdptype), ('state', np.uint32)])
+        self.streamdtype = np.dtype([('vid', '<i4'), ('ts', fdptype),
+                                          ('te', fdptype)])
+        self.eventbuffdtype = np.dtype([('tinit', fdptype), ('state', np.uint32),
+                                    ('ts', fdptype), ('te', fdptype)])
+
+        gamma = self.cfg.getfloat('constants', 'gamma')
+        avgrho = self.cfg.getfloat(cfgsect, 'avg-rho')
+        avgmach = self.cfg.getfloat(cfgsect, 'avg-mach')
+        ti = self.cfg.getfloat(cfgsect, 'turbulence-intensity')
+        sigma = self.cfg.getfloat(cfgsect, 'sigma')
+        centre = self.cfg.getliteral(cfgsect, 'centre')
+        rotaxis = np.array(self.cfg.getliteral(cfgsect, 'rot-axis'))
+        rotangle = -1.0*np.radians(self.cfg.getfloat(cfgsect,'rot-angle'))
+        self.avgu = avgu = self.cfg.getfloat(cfgsect, 'avg-u')
+        self.ls = ls = self.cfg.getfloat(cfgsect, 'turbulence-length-scale')
+
+        gc = (2*sigma/(math.erf(1/sigma)*math.pi**0.5))**0.5
+        fac1 = -0.5/(sigma*ls)**2
+        fac2 = avgrho*(gamma - 1)*avgmach**2
+        fac3 = 0.01*ti*avgu*(gc/sigma)**3
+
+        self.ydim = ydim = self.cfg.getfloat(cfgsect, 'y-dim')
+        self.zdim = zdim = self.cfg.getfloat(cfgsect, 'z-dim')
+        
+        self.ymin = ymin = -ydim/2
```

Comment:
I'd nix y/z and instead just use a two element list.  So `self.yzmin = -0.5*np.array([ydim, zdim])`

Reply:
Just following up here.